drivers/flash/flash_sam0.c

```diff
@@ -163,10 +165,11 @@ static int flash_sam0_write_page(const struct device *dev, off_t offset,
 	flash_sam0_wait_ready();
 
 	/* Ensure writes happen 32 bits at a time. */
-	for (; src != end; src++, dst++) {
-		*dst = UNALIGNED_GET((uint32_t *)src);
+	for (; len > 0; len -= sizeof(*src)) {
```

Comment:
Minor concern: add a check or at least an assert that `len % sizeof(*src) == 0U` to as a sanity check to prevent an infinite loop.

Reply:
Indeed, what's wrong with the original? (this change will fail and `len` will wrap to "_very positive_" if its not a multiple of 4...)

Reply:
It's handled in [`flash_sam0_write()`](https://github.com/zephyrproject-rtos/zephyr/blob/19ef21d91cd11a905e2460952556a1da7811877d/drivers/flash/flash_sam0.c#L297) using a multiple of `FLASH_WRITE_BLK_SZ`, but this pushes the relevant checking / info out-of-file and into the device trees...

Reply:
The idea was to mirror as much as possible the method implemented in flash_sam.c.  The same risk would exist in that code as well...